plugins/processors/cumulative_sum/cumulative_sum.go

```diff
@@ -0,0 +1,104 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package cumulative_sum
+
+import (
+	_ "embed"
+	"fmt"
+	"maps"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type CumulativeSum struct {
+	Fields         []string        `toml:"fields"`
+	ExpiryInterval config.Duration `toml:"expiry_interval"`
+	Log            telegraf.Logger `toml:"-"`
+	accept         filter.Filter
+	cache          map[uint64]*entry
+}
```

Comment:
Separate the internal fields from the user-settings...
```suggestion
type CumulativeSum struct {
	Fields         []string        `toml:"fields"`
	ExpiryInterval config.Duration `toml:"expiry_interval"`
	Log            telegraf.Logger `toml:"-"`

	accept filter.Filter
	cache  map[uint64]*entry
}
```